Crash on MdbxTx == nil, when it is accessed after Rollback()

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1055d5b3c]

goroutine 94 [running]:
github.com/erigontech/mdbx-go/mdbx.openCursor.func1(0x0, 0x69, 0x14000912760)
        /Users/blade/go/pkg/mod/github.com/erigontech/mdbx-go@v0.27.24/mdbx/cursor.go:69 +0x2c
github.com/erigontech/mdbx-go/mdbx.openCursor(0x0, 0x69)
        /Users/blade/go/pkg/mod/github.com/erigontech/mdbx-go@v0.27.24/mdbx/cursor.go:69 +0xb0
github.com/erigontech/mdbx-go/mdbx.(*Txn).OpenCursor(0x0, 0x69)
        /Users/blade/go/pkg/mod/github.com/erigontech/mdbx-go@v0.27.24/mdbx/txn.go:597 +0x3c
github.com/ledgerwatch/erigon-lib/kv/mdbx.(*MdbxTx).stdCursor(0x14002a5e600, {0x1071bf62c, 0xa})
        /Users/blade/work/software/cdk-erigon/erigon-lib/kv/mdbx/kv_mdbx.go:1352 +0x220
github.com/ledgerwatch/erigon-lib/kv/mdbx.(*MdbxTx).RwCursor(0x14002a5e600, {0x1071bf62c, 0xa})
        /Users/blade/work/software/cdk-erigon/erigon-lib/kv/mdbx/kv_mdbx.go:1332 +0xa0
github.com/ledgerwatch/erigon-lib/kv/mdbx.(*MdbxTx).statelessCursor(0x14002a5e600, {0x1071bf62c, 0xa})
        /Users/blade/work/software/cdk-erigon/erigon-lib/kv/mdbx/kv_mdbx.go:1196 +0x11c
github.com/ledgerwatch/erigon-lib/kv/mdbx.(*MdbxTx).GetOne(0x14002a5e600, {0x1071bf62c, 0xa}, {0x1400171fd58, 0x14, 0x14})
        /Users/blade/work/software/cdk-erigon/erigon-lib/kv/mdbx/kv_mdbx.go:1222 +0x5c
github.com/ledgerwatch/erigon-lib/kv/kvcache.(*DummyCache).Get(0x1090fa0c0, {0x1400171fd58, 0x14, 0x14}, {0x107dbf210, 0x14002a5e600}, 0x0)
        /Users/blade/work/software/cdk-erigon/erigon-lib/kv/kvcache/dummy.go:39 +0x7c
github.com/ledgerwatch/erigon-lib/kv/kvcache.(*DummyView).Get(0x14002cf9488, {0x1400171fd58, 0x14, 0x14})
        /Users/blade/work/software/cdk-erigon/erigon-lib/kv/kvcache/dummy.go:53 +0x84
github.com/ledgerwatch/erigon/zk/txpool.(*sendersBatch).info(0x14001c0ac60, {0x107d9f8a8, 0x14002cf9488}, 0x8)
        /Users/blade/work/software/cdk-erigon/zk/txpool/pool.go:2066 +0x12c
github.com/ledgerwatch/erigon/zk/txpool.(*TxPool).RemoveMinedTransactions(0x140005c8a08, {0x107daa400, 0x14001264c30}, {0x107dbf210, 0x14002a5e600}, 0x4000000000000, {0x140013c8000, 0x1c8, 0x200})
        /Users/blade/work/software/cdk-erigon/zk/txpool/pool_zk.go:331 +0x4bc
github.com/ledgerwatch/erigon/zk/stages.sequencingBatchStep(_, {_, _}, {_, _}, {{0x107db7650, 0x14001986480}, 0x10000000, {0x1, {0x107dad600, ...}, ...}, ...}, ...)
        /Users/blade/work/software/cdk-erigon/zk/stages/stage_sequence_execute.go:700 +0x2ad4
github.com/ledgerwatch/erigon/zk/stages.SpawnSequencingStage(_, {_, _}, {_, _}, {{0x107db7650, 0x14001986480}, 0x10000000, {0x1, {0x107dad600, ...}, ...}, ...}, ...)
        /Users/blade/work/software/cdk-erigon/zk/stages/stage_sequence_execute.go:80 +0x6d4
github.com/ledgerwatch/erigon/zk/stages.SequencerZkStages.func13(0x0, 0x0, 0x140006708a0, {0x107da1108, 0x140000fdb80}, {{0x0, 0x0}, {0x0, 0x0}}, {0x107db69e8, ...})
        /Users/blade/work/software/cdk-erigon/zk/stages/stages.go:97 +0xdc
github.com/ledgerwatch/erigon/eth/stagedsync.(*Sync).runStage(0x140000fdb80, 0x14001c26050, {0x107db7650, 0x14001986480}, {{0x0, 0x0}, {0x0, 0x0}}, 0x0, 0x0)
        /Users/blade/work/software/cdk-erigon/eth/stagedsync/sync.go:513 +0x1c4
github.com/ledgerwatch/erigon/eth/stagedsync.(*Sync).Run(0x140000fdb80, {0x107db7650, 0x14001986480}, {{0x0, 0x0}, {0x0, 0x0}}, 0x0)
        /Users/blade/work/software/cdk-erigon/eth/stagedsync/sync.go:383 +0x464
github.com/ledgerwatch/erigon/turbo/stages.StageLoopIteration({0x107daa400, 0x14001264c30}, {0x107db7650, 0x14001986480}, {{0x0, 0x0}, {0x0, 0x0}}, 0x140000fdb80, 0x0, ...)
        /Users/blade/work/software/cdk-erigon/turbo/stages/stageloop.go:154 +0x554
github.com/ledgerwatch/erigon/turbo/stages.StageLoop({0x107daa400, 0x14001264c30}, {0x107db7650, 0x14001986480}, 0x140000fdb80, 0x14000034c00, 0x140019f4070, 0x0, {0x107db69e8, 0x140001ec820}, ...)
        /Users/blade/work/software/cdk-erigon/turbo/stages/stageloop.go:73 +0x180
created by github.com/ledgerwatch/erigon/eth.(*Ethereum).Start in goroutine 1
        /Users/blade/work/software/cdk-erigon/eth/backend.go:1909 +0x740
```

--- a/zk/stages/stage_sequence_execute.go
+++ b/zk/stages/stage_sequence_execute.go
@@ -695,11 +695,7 @@ func sequencingBatchStep(
 		return err
 	}
 
-	if err = cfg.txPool.RemoveMinedTransactions(ctx, sdb.tx, header.GasLimit, minedTxsToRemove); err != nil {
-		return err
-	}
-
-	return nil
+	return removeMinedTransactionsFromPool(ctx, cfg, header.GasLimit, minedTxsToRemove, logPrefix)
 }
 
 func removeInclusionTransaction(orig []types.Transaction, index int) []types.Transaction {
@@ -712,4 +708,18 @@ func removeInclusionTransaction(orig []types.Transaction, index int) []types.Tra
 func isOkKnownError(err error) bool {
 	return err == nil ||
 		errors.Is(err, core.ErrNonceTooHigh)
+}
+
+func removeMinedTransactionsFromPool(ctx context.Context, cfg SequenceBlockCfg, gasLimit uint64, minedTxsToRemove []common.Hash, logPrefix string) error {
+	roTx, err := cfg.db.BeginRo(ctx)
+	if err != nil {
+		return err
+	}
+	defer roTx.Rollback()
+
+	if err = cfg.txPool.RemoveMinedTransactions(ctx, roTx, gasLimit, minedTxsToRemove); err != nil {
+		return fmt.Errorf("[%s] removing mined transactions from pool: %w", logPrefix, err)
+	}
+
+	return nil
 }

